File: frameworks/java/spring-boot/Implementation/src/main/java/implementation/routes/CompressedRoutes.java
package implementation.routes;

import implementation.Payload;
import implementation.Payloads;
import implementation.Serial;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * compressed: these routes answer like any other. Tomcat's compression, which
 * application.properties turns on for the whole connector, gzips the answer when the request asks
 * for it.
 */
@RestController
public class CompressedRoutes {

    private final Payloads p;

    CompressedRoutes(Payloads p) {
        this.p = p;
    }

    @GetMapping("/compressed/small")
    public Payload small(HttpServletResponse response) {
        Serial.write(response);
        return p.small();
    }

    @GetMapping("/compressed/large")
    public Payload large(HttpServletResponse response) {
        Serial.write(response);
        return p.large();
    }
}

File: frameworks/dotnet/minimal-apis/Implementation/Routes/CompressedRoutes.cs
namespace Implementation.Routes;

/// <summary>
/// compressed: these routes answer like any other. ASP.NET Core's response compression,
/// installed on the whole application, gzips the answer when the request asks for it.
/// </summary>
public static class CompressedRoutes
{
    public static void Map(IEndpointRouteBuilder app)
    {
        app.MapGet("/compressed/small", (HttpResponse response, Payloads p) => Fresh(response, p.Small)).DisableValidation();

        app.MapGet("/compressed/large", (HttpResponse response, Payloads p) => Fresh(response, p.Large)).DisableValidation();
    }

    private static Payload Fresh(HttpResponse response, Payload payload)
    {
        Serial.Write(response);
        return payload;
    }
}
